extensions/amp-story-panning-media/0.1/amp-story-panning-media.js

```diff
@@ -32,55 +35,112 @@ export class AmpStoryPanningMedia extends AMP.BaseElement {
     /** @private {?Element} */
     this.element_ = element;
 
-    /** @private {?Element} */
-    this.image_ = null;
+    /** @private {?Element} The element that is transitioned. */
+    this.ampImgEl_ = dev().assertElement(
+      this.element_.querySelector('amp-img')
+    );
 
-    /** @private {?string} */
-    this.x_ = null;
+    /** @private {?string} Sent to siblings to update their position. */
+    this.x = this.element_.getAttribute('x') || '0%';
 
-    /** @private {?string} */
-    this.y_ = null;
+    /** @private {?string} Sent to siblings to update their position. */
+    this.y = this.element_.getAttribute('y') || '0%';
 
-    /** @private {?string} */
-    this.zoom_ = null;
+    /** @private {?string} Sent to siblings to update their position. */
+    this.zoom = this.element_.getAttribute('zoom') || '1';
+
+    /** @private {Array<Element>} */
+    this.siblings_ = [];
   }
 
   /** @override */
   buildCallback() {
-    this.x_ = this.element_.getAttribute('x') || '0%';
-    this.y_ = this.element_.getAttribute('y') || '0%';
-    this.zoom_ = this.element_.getAttribute('zoom') || '1';
+    this.getSiblings_().then((siblings_) => {
+      this.siblings_ = siblings_;
+    });
+
+    return Services.storyStoreServiceForOrNull(this.win).then(
+      (storeService) => {
+        storeService.subscribe(StateProperty.CURRENT_PAGE_ID, (currPageId) => {
+          const isOnActivePage = currPageId === this.getPageId_();
+          if (isOnActivePage) {
+            this.updateSiblings_();
+          }
+        });
+      }
+    );
   }
 
   /** @override */
   layoutCallback() {
-    const ampImgEl = dev().assertElement(
-      this.element_.querySelector('amp-img')
-    );
-    return whenUpgradedToCustomElement(ampImgEl)
-      .then(() => ampImgEl.signals().whenSignal(CommonSignals.LOAD_END))
+    return whenUpgradedToCustomElement(this.ampImgEl_)
+      .then(() => this.ampImgEl_.signals().whenSignal(CommonSignals.LOAD_END))
       .then(() => {
-        this.image_ = dev().assertElement(this.element.querySelector('img'));
+        const imgEl = dev().assertElement(this.element_.querySelector('img'));
         // Remove layout="fill" classes so image is not clipped.
-        this.image_.classList = '';
-        // Fill image to 100% height of viewport.
+        imgEl.classList = '';
+        // Centers the amp-img horizontally. The image does not load if this is done in CSS.
         // TODO(#31515): Handle base zoom of aspect ratio wider than image
-        setStyles(this.image_, {height: '100%'});
-        return this.updatePosition_();
+        setStyles(this.ampImgEl_, {
+          left: 'auto',
+          right: 'auto',
+        });
+        return Promise.resolve();
       })
       .catch(() => user().error(TAG, 'Failed to load the amp-img.'));
   }
 
   /**
+   * @return {!Promise<Array<Element>>}
+   */
+  getSiblings_() {
+    // TODO Group siblings to be transitioned together #31932
```

Comment:
```suggestion
    // TODO(#31932): Group siblings to be transitioned together.
```

nit: use standard TODO notation